### DISCUSSION TITLE
Q about not detecting Passer domesticus?

### DISCUSSION THREAD

**Discussion by unosonic** [mcguirepr89/BirdNET-Pi#358] [Q&A] [answered]
hi there, new user here... last night I've found BirdNet-Pi, now it's running in my backyard! (Pi3, IQAudio Codec plus hat, AOM 5024 mics) Thanks for all the great work! 
First question: BirdNet-pi has correctly detected swifts and green finches immediately, but not the 3 House sparrows chirping all the time next to the mics ;) How comes? They are not on the excluded species list, as far as I can see on the web interface. Thanks for a hint!
![grafik](https://user-images.githubusercontent.com/54538102/173026951-5f542b06-9982-4067-9cc1-b023b00f95c7.png)

**Comment by ehpersonal38** [comment]
Next time the House Sparrows are chirping, I would recommend going to "View Log" on your BirdNET-Pi interface, and look for the confidence values the model is returning. You'll see something like:

```
13:35:37---0.0;3.0-("Passer domesticus_House Sparrow", 0.13170013)
13:35:37---3.0;6.0-('Passer domesticus_House Sparrow', 0.14684723)
13:35:37---6.0;9.0-('Passer domesticus_House Sparrow', 0.13127398)
13:35:37---9.0;12.0-("Passer domesticus_House Sparrow", 0.14561795)
13:35:37---12.0;15.0-("Passer domesticus_House Sparrow", 0.13758358)
```

Note the average confidence values - here it's around 0.13. Then you can go to `Tools -> Settings -> Advanced Settings -> Minimum Confidence` and enter in 0.13 (or a bit lower). You should then be getting House Sparrow detections.

**Reply by Klages** [reply]
in my environment the birds sing almost continuously, with a security of .5 i have about 250 detections a day. if i set the security to .1 i have matches every second, which also apply to birds that can be found here. can it be that the security is low because my microphone is too bad? would there be a higher security on the matches with a better microphone?

Comapring spectrograms from a Haussperling at 76% and 12%
![image](https://user-images.githubusercontent.com/3393617/173348919-b5d0f3a6-51f5-41d2-a70a-e9301633dab3.png)
![image](https://user-images.githubusercontent.com/3393617/173348931-f48ede45-843e-410f-8619-ce2a6ff67e2e.png)

**Reply by unosonic** [reply]
In general I'd think that a low noise microphon and a reasonable high gain gives better results, even for more distant calls/songs. If your signal to noise ratio is bad, that would of course affect recognition. Both Primo EM *72 and AOM 5024 mics have a low self noise.

**Reply by ehpersonal38** [reply]
Out of curiosity I did some tests with noise. I analyzed a single audio clip of a Carolina Chickadee with 2 levels of noise added.

### Original Clip:
![origspectrogram](https://user-images.githubusercontent.com/103586016/173379186-067a3b31-46ee-4782-ace9-d5c9c15c6194.png)

**Result: Carolina Chickadee, confidence 0.78**

### Low Noise Added:
![test2spectrogram](https://user-images.githubusercontent.com/103586016/173379316-5394c409-358b-4d04-a1a3-393e8c2f848f.png)

**Result: Carolina Chickadee, confidence 0.69** 

### High Noise Added:
![test3spectrogram](https://user-images.githubusercontent.com/103586016/173380565-53ccef3b-1a05-4595-b529-f92f853dc03e.png)

**Result: Carolina Chickadee, confidence 0.58** 

Based on this one trial, the model is surprisingly resilient to noise! 

@mcguirepr89 you might be interested in this^

**Reply by mcguirepr89** [reply]
@ehpersonal38 Thanks!

**Reply by unosonic** [reply]
that's indeed interesting, I would have expected lesser confidence. good to know.
btw the sparrows in my backyard have moved, now they seem to sit in a pleasant distance to the microphones without much reverb from walls, and voila, confidence rises up to 85%

**Comment by unosonic** [comment]
I have a tail -f daemon.log running all the time while watching/listening the spectrogram, and I did play with the threshold. Of course it works then, but at a confidence of <= 0.2 I get an amazing spectrum of (phantom) biodiversity here in my backyard ;-)
Interesting thing is: it detects almost any swift rushing by with an confidence of 80% and more, even if it's far away. So nice feature btw, that it prints the name of recognized birds onto the spectrogram area in question. 
I wonder whether some echoing in that rather narrow yard may have a negative effect on recognition. On the other hand, that's also what we hear sometimes in forest.

**Comment by unosonic** [comment, marked as answer]
Since I've asked this Q flocks of house sparrows have been recognized, so I think it's ok to close this ticket.
